dcos/dcos-ui
# DCOS-48333: Jobs form schedule section

This PR adds the schedule tab and form section to the jobs form. See intended designs [here](https://mesosphere.invisionapp.com/share/Y6Q8EFKBA3G#/screens/345675529). 

## ⚠️Dependencies

This PR is based on #3635 and can only be merged after the other one got merged
It makes use of some new Error utils and structural changes.



## Testing

Try testing the form validation and editing via JSON editor. Some specific validation scenarios:

Schedule is blank if form looks blank:
- Go to schedule tab
- Toggle either or both of the checkboxes on and off
- Submit form and ensure there are no schedule errors

Similar to above:
- Go to schedule tab
- enter an ID
- Toggle either or both of the checkboxes on and off
- Delete ID
- Submit form and ensure there are no schedule errors

## Trade-offs

Building on patterns established in #3599 and #3635
